generic-tabs/GenericTabs.js

```diff
@@ -71,34 +77,18 @@ export class GenericTabs extends SelectedMixin(BatchingElement) {
   constructor() {
     super();
 
-    // Check if slots were setup
-    // If not, assume heading structure and preprocess light DOM
-    if (!this.querySelector('[slot]')) {
-      const unformattedEls = [...this.children];
-      unformattedEls.forEach(node => {
-        if (node instanceof HTMLHeadingElement) {
-          const button = document.createElement('button');
-          button.textContent = node.textContent;
-          button.setAttribute('slot', 'tab');
-          node.parentNode.replaceChild(button, node);
-        } else {
-          node.setAttribute('slot', 'panel');
-        }
-      });
-    }
-
     this.attachShadow({ mode: 'open' });
     this.shadowRoot.appendChild(template.content.cloneNode(true));
   }
 
   update() {
-    // Preprocess markup for headings
     const { tabs, panels } = this.getElements();
     tabs.forEach((_, i) => {
+      tabs[i].slot = 'tab';
       if (i === this.selected) {
         tabs[i].setAttribute('selected', '');
         tabs[i].setAttribute('aria-selected', 'true');
-        tabs[i].removeAttribute('tabindex');
+        tabs[i].setAttribute('tabindex', '0');
```

Comment:
This change is required to support focusable headings. It has little effect on the classic buttons, who would previous have the `tabindex` added and removed. The component adds `role="tab"` and keyboard navigation to the heading, but I’m not sure if this good enough for accessibility.

In VoiceOver, the interactive command worked to select an unselected tab. However, on a selected button the spacebar would have no effect, whereas on these headings the spacebar performs the default scroll operation. If that’s important, then I think this issue would be better solved in the parent component, as a separate issue, since it’s an existing issue not introduced by this PR.